network/forward.go

```diff
@@ -0,0 +1,130 @@
+package network
+
+import (
+	"reflect"
+
+	"github.com/xh3b4sd/anna/api"
+	"github.com/xh3b4sd/anna/key"
+	"github.com/xh3b4sd/anna/spec"
+)
+
+// receiver
+
+func (n *network) forwardCLGs(ctx spec.Context, behaviorIDs []string, payload spec.NetworkPayload) error {
+	for _, ID := range behaviorIDs {
+		// Prepare a new context for the new connection path.
+		newCtx := ctx.Clone()
+		newCtx.SetBehaviorID(ID)
+
+		// Create a new network payload. Note that the old context of the old
+		// network payload is removed to only append actual arguments to the new
+		// network payload.
+		newPayloadConfig := api.DefaultNetworkPayloadConfig()
+		newPayloadConfig.Args = append([]reflect.Value{reflect.ValueOf(newCtx)}, payload.GetArgs()[1:]...)
+		newPayloadConfig.Destination = spec.ObjectID(ID)
+		newPayloadConfig.Sources = []spec.ObjectID{payload.GetDestination()}
+		newPayload, err := api.NewNetworkPayload(newPayloadConfig)
+		if err != nil {
+			return maskAny(err)
+		}
+
+		// Find the actual CLG based on its behavior ID. Therefore we lookup the
+		// behavior name by the given behavior ID. Data we read here is written
+		// within several CLGs. That way the network creates its own connections
+		// based on learned experiences.
+		//
+		// TODO where are these connections coming from?
+		// TODO if there are none, we need to find some randomly
+		// TODO there needs to be some sort of variation when executing existing CLG trees
+		//
+		clgName, err := n.Storage().General().Get(key.NewCLGKey("behavior-id:%s:behavior-name", ID))
+		if err != nil {
+			return maskAny(err)
+		}
+		CLG, err := n.clgByName(clgName)
+		if err != nil {
+			return maskAny(err)
+		}
+		CLG.GetInputChannel() <- newPayload
+	}
+
+	return nil
+}
+
+// TODO
+func (n *network) forwardInputCLG(payload spec.NetworkPayload) error {
+	ctx, err := payload.GetContext()
+	if err != nil {
+		return maskAny(err)
+	}
+
+	// TODO fetch CLG tree ID using information ID
+	// TODO fetch input CLG ID (behaviorID) using CLG tree ID
```

Comment:
We need to lookup the behavior ID of the input CLG of the current CLG tree. Using this behavior ID we can forward the network payload from the output CLG to the input CLG.